pkgs/applications/misc/auto-multiple-choice/default.nix

```
@@ -23,14 +23,16 @@
 , poppler
 , auto-multiple-choice
 }:
-stdenv.mkDerivation rec {
+lib.fix (self: stdenv.mkDerivation rec {
   pname = "auto-multiple-choice";
   version = "1.5.2";
   src = fetchurl {
     url = "https://download.auto-multiple-choice.net/${pname}_${version}_precomp.tar.gz";
     sha256 = "sha256-AjonJOooSe53Fww3QU6Dft95ojNqWrTuPul3nkIbctM=";
   };
-  tlType = "run";
+
+  passthru.tlType = "run";
+  passthru.pkgs = [ self ];
```

```suggestion
  passthru = {
    tlType = "run";
    pkgs = [ self ];
  };
```
I doubt that we want to passthru self as pkgs here

---

The derivation cannot refer to itself so it has to be passthru in this case.

---

(Indeed, 'infinite recursion ecountered' after a quick try. Also, `pkgs` and `tlType` are not used in the builder environment, so passthru makes sense for this. Marking as resolved.)